src/ports/rt-kernel/pnal_snmp.c

```diff
@@ -0,0 +1,109 @@
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2020 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "options.h"
+#include "pnet_api.h"
+#include "osal.h"
+#include "osal_log.h"
+#include "pnal.h"
+#include "pf_types.h"
+#include "pnal_snmp.h"
+
+#include "lldp-mib.h"
+
+#include <lwip/apps/snmp.h>
+#include <lwip/apps/snmp_mib2.h>
+#include <lwip/apps/snmp_threadsync.h>
+
+/* Global variable for SNMP server state */
+pnal_snmp_t pnal_snmp;
```

Comment:
A while ago we spent some effort making sure several instances of p-net could exist simultaneously. This would again limit the stack to one instance (if SNMP is used)? Is there no way to associate the snmp state with the p-net instance?

Reply:
The lwip SNMP stack does not support more than one server instance. If more than one p-net stack is running on rt-kernel, then only one of them may use SNMP.
Associating the SNMP state with the p-net instance is furthermore not supported by lwip, as the callback functions used for implementing the MIBs does not supply any opaque pointer.